pkg/costmodel/allocation.go

```diff
@@ -252,6 +257,26 @@ func (cm *CostModel) ComputeAllocation(start, end time.Time, resolution time.Dur
 	return result, nil
 }
 
+// DateRange checks the data (up to 90 days in the past), and returns the oldest and newest sample timestamp from opencost scraping metric
+// it supposed to be a good indicator of available allocation data
+func (cm *CostModel) DateRange() (time.Time, time.Time, error) {
```

Comment:
I'm wondering if a new metric here makes sense vs. just using an opencost specific metric, for example node_total_hourly_cost or similar to check the oldest/newest samples?

@nikovacevic I'm not the expert here... any thoughts?

Reply:
I don't think it's matter which metric is used in this case. Can be anything.

`up` is prometheus built-in metric. It's sampled on every scrape. No scape = no data.

https://prometheus.io/docs/concepts/jobs_instances/#automatically-generated-labels-and-time-series


It can be further generalized:
```
max_over_time(timestamp(group(up))[1d:10h])
```

Reply:
Totally understand; I think some managed promQL systems do not have an "up" metric necessarily-- so maybe one that is emitted by the cost-model makes the most sense?

Reply:
Is there an opencost metric that is guaranteed to be produced?

Reply:
yes-- sorry for the late response. it's container_memory_allocation_bytes https://github.com/opencost/opencost/blob/13f6525c09fd9427d18afea5efb26b3e27c7866d/pkg/costmodel/metrics.go#L205